pkg/storage/stores/indexshipper/shipper.go

```diff
@@ -139,11 +139,15 @@ func (s *indexShipper) AddIndex(tableName, userID string, index index.Index) err
 
 func (s *indexShipper) ForEach(ctx context.Context, tableName, userID string, callback func(index index.Index) error) error {
 	if s.downloadsManager != nil {
-		return s.downloadsManager.ForEach(ctx, tableName, userID, callback)
+		if err := s.downloadsManager.ForEach(ctx, tableName, userID, callback); err != nil {
```

Comment:
I thought this was by design: only using either the downloads mgr or the uploads mgr so we didn't query the same indices in `R/W` mode for the shipper.